traits/ctraits.c

```diff
@@ -808,10 +808,18 @@ static void
 has_traits_dealloc(has_traits_object *obj)
 {
     PyObject_GC_UnTrack(obj);
+#if PY_VERSION_HEX < 0x03080000
```

Comment:
We can simplify this if we drop support for Python 3.7, but doing that in this PR would be unnecessarily complicated. I've opened a separate issue #1769.